src/components/CompareResults/CompareResultsTable.tsx

```diff
@@ -6,52 +10,237 @@ import TableContainer from '@mui/material/TableContainer';
-              <CompareResultsTableRow
-                key={index}
-                result={result}
-                index={index}
-                mode={mode}
-              />
-            ))}
-        </TableBody>
-      </Table>
-    </TableContainer>
+    <React.Fragment>
+      <Autocomplete
+        id="size-small-standard"
+        size="small"
+        sx={{ width: 200, marginBottom: 5 }}
+        options={frameworks}
+        isOptionEqualToValue={(option, value) => option.name === value.name}
+        getOptionLabel={(option) => option.name}
+        defaultValue={frameworks[0]}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            variant="standard"
+            label="Framework"
+            placeholder="Select framework"
+          />
+        )}
+      />
+      {isFiltered && <Alert severity="info">Filters have been applied</Alert>}
+      <TableContainer component={Paper}>
+        <Table sx={{ minWidth: 650 }} size="small" aria-label="a dense table">
+          <TableHead>
+            <TableRow>
+              {tableHead.map(
+                ({ label, key, align }: CompareResultsTableHeader, index) => {
+                  let options: string[] | ConfidenceValue[] = [];
+                  let id = '';
+                  const headerId = key as keyof typeof filter;
+                  if (filterKeys.includes(headerId)) {
+                    options = Array.from(filter[headerId].options.values());
+                    id = open ? `${key as keyof typeof filter}` : '';
+                  }
+
+                  return (
+                    <TableCell key={index} align={align}>
+                      {label}
+                      {filterKeys.includes(headerId) && (
+                        <React.Fragment>
+                          <IconButton
+                            data-testid={`${headerId}-options-button`}
+                            onClick={(e) => handleOpenOptions(e, headerId)}
+                          >
+                            <MoreVertIcon />
+                          </IconButton>
+                          <Popover
+                            id={id}
+                            open={openFilter === id}
+                            anchorEl={anchorEl}
+                            onClose={handleCloseOptions}
+                            anchorOrigin={{
+                              vertical: 'bottom',
+                              horizontal: 'center',
+                            }}
+                            transformOrigin={{
+                              vertical: 'top',
+                              horizontal: 'right',
+                            }}
+                          >
+                            <FilterOptionsList
+                              options={options}
+                              column={headerId}
+                            />
+                          </Popover>
+                        </React.Fragment>
```

Comment:
From our yesterday discussion, we've seen that it would be good for performance (and probably for readability as well) to extract this part to a separate component, so that any change to the state of the popover wouldn't render the full table again.